src/entity_builder.rs

```diff
@@ -374,15 +387,24 @@ impl<M> Default for Common<M> {
 }
 
 #[derive(Clone)]
-struct Cloneable(unsafe fn(*const u8, &mut dyn FnMut(*mut u8, TypeInfo)));
+/// Struct providing dynamic clone via callback
+pub struct Cloneable {
+    func: unsafe fn(*const u8, TypeInfo, &mut dyn FnMut(*mut u8, TypeInfo)),
+    // Store type info along with cloneable
+    type_info: TypeInfo,
```

Comment:
Why the change to store this rather than constructing it on-demand?

Reply:
You are right. If typeinfo is cheap to construct it is not necessary to store it.

Reply:
It was left over from storing the type info in an earlier attempt. Since it is known with bundle::put I thought to store it rather than compute again.

Reply:
The function is from my understanding called for each build

Reply:
`TypeInfo::of` will almost certainly be compiled out into a constant, so storing `TypeInfo`s will probably *add* overhead, in addition to complicating the code. As usual, beware premature optimizations.